### PR TITLE
Create proxy_susp_ipfs_cred_harvest.yml

### PR DESCRIPTION
**Pull request by m4nbat**
### Summary of the Pull Request

This pull request is to propose a new rule that can be used to hunt for possible post phishing communication with a credential harvesting page hosted on the interplanetary file system (IPFS)

### Detailed Description of the Pull Request / Additional Comments

The intent of the rule is to look for network connections to IPFS via the proxy where the URI contains the characteristics observed in recent post phishing activity. The rule was tested with a Microsoft Sentinel conversion.

### Example Log Event

Url in logs:

hxxpsx//ipfs.io/ipfs/QmSejSajLFEhYUHT6v45tbtajmRzJf5XTrv9kMA1?filename=next.hum.html#me@me.com
hxxpsx//ipfs.io/ipfs/QmbQopz8dyJFHszxty67ui8CMx1mqCp85neyL6D?filename=P%20O%20NN%20Y%20TT%20R%20SS.html#bill.bob@b.com

Example IPFS URI

/ipfs/QmbQopz8dyJFHszxty67ui8CMx1mqCp85neyL6D?filename=P%20O%20NN%20Y%20TT%20R%20SS.html#bill.bob@b.com

### Fixed Issues

N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

File: rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+title: Suspicious Network Communication With IPFS
+id: eb6c2004-1cef-427f-8885-9042974e5eb6
+status: experimental
+description: Detects connections to interplanetary file system (IPFS) containing a users email address which mirrors behaviours observed in recent phishing campaigns leveraging IPFS to host credential harvesting webpages.
```
Comment (phantinuss) [inline rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml:4]:
```suggestion
description: Detects connections to interplanetary file system (IPFS) containing a user's email address which mirrors behaviours observed in recent phishing campaigns leveraging IPFS to host credential harvesting webpages.
```

File: rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    - attack.credential_access
+    - attack.t1056
+logsource:
+    category: proxy
+detection:
+    selection:
+        cs-uri|re: '(?i)ipfs.io/ipfs.+\..+@.+\..+'
+    condition: selection
+falsepositives:
+    - Legitimate use of IPFS being used in the organiation. However the cs-uri regex looking for a user email will likely negate this.
```
Comment (phantinuss) [inline rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml:21]:
```suggestion
    - Legitimate use of IPFS being used in the organisation. However the cs-uri regex looking for a user email will likely negate this.
```

File: rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+author: Gavin Knapp
+date: 2023/03/16
+tags:
+    - attack.credential_access
+    - attack.t1056
+logsource:
+    category: proxy
+detection:
+    selection:
+        cs-uri|re: '(?i)ipfs.io/ipfs.+\..+@.+\..+'
```
Comment (phantinuss) [inline rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml:18]: which of these dots and `\` should be literal and which are used for escaping/matching any character?
I assume at least the dot in `ipfs.io` needs to be escaped

Comment (m4nbat) [reply rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml:18]: Oops apologies, not enough coffee yesterday, hopefully this one is accurate:

(?i)(ipfs\.io/|ipfs\.io\s).+\..+@.+\.[a-zA-Z]+

image

**Review by phantinuss — APPROVED** [review]
looks good to me. As long as it is a low rule :)

File: rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+title: Suspicious Network Communication With IPFS
+id: eb6c2004-1cef-427f-8885-9042974e5eb6
+status: experimental
+description: Detects connections to interplanetary file system (IPFS) containing a user's email address which mirrors behaviours observed in recent phishing campaigns leveraging IPFS to host credential harvesting webpages.
+references:
+    - https://blog.talosintelligence.com/ipfs-abuse/
+    - https://github.com/Cisco-Talos/IOCs/tree/main/2022/11
```
Comment (nasbench) [inline rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml:7]:
```suggestion
    - https://github.com/Cisco-Talos/IOCs/tree/80caca039988252fbb3f27a2e89c2f2917f582e0/2022/11
```

**Review by nasbench — APPROVED** [review]

**Pull request merged**